modules/serverless-common/pages/service-mesh/common-service-mesh-network-isolation.adoc

```diff
@@ -0,0 +1,438 @@
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+. With this set up in place, cluster administrators can use their own `AuthorizationPolicies` to define which namespaces (tenants) can communicate with each other. Each namespace needs:
+- One AuthorizationPolicy limiting directly incoming traffic in the tenants namespace
+- One AuthorizationPolicy limiting incoming traffic via activator in the `knative-serving` namespace
+- One AuthorizationPolicy allowing Kubernetes to call PreStopHooks on Knative Services
++
+As it is a cumbersome task to create all those policies by hand, you can use our link:https://github.com/openshift-knative/knative-istio-authz-chart[helm based generator] to create the necessary resources for each tenant:
++
+[source,terminal]
+.Create resources per tenant with helm
+----
+helm template oci://quay.io/openshift-knative/knative-istio-authz-onboarding --version 1.31.0 --set "name=tenant-1" --set "namespaces={ns1, ns2}"
+helm template oci://quay.io/openshift-knative/knative-istio-authz-onboarding --version 1.31.0 --set "name=tenant-2" --set "namespaces={ns3, ns4}"
+----
++
+And apply the generated resources to your cluster:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
++
+[NOTE]
+====
+The helm chart has several options that can be passed to configure the generated resources. Please refer to the link:https://github.com/openshift-knative/knative-istio-authz-chart/blob/main/values.yaml[values.yaml] for a full reference.
+====
+
+
+.Verifying the configuration
+
+This verification is assuming that we have two tenants with one namespace each, all part of the `ServiceMeshMemberRoll`, configured with resources listed above.
+We can then use curl to verify the connectivity:
+
+. Deploy Knative Services in both tenants namespaces and a `curl` pod to run test commands:
++
+[source,terminal]
+----
+# Tenant 1
+cat <<-EOF | oc apply -f -
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: test-webapp
+  namespace: tenant-1
+  annotations:
+    serving.knative.openshift.io/enablePassthrough: "true"
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: 'true'
+        autoscaling.knative.dev/target-burst-capacity: "-1"
+    spec:
+      containers:
+        - image: docker.io/openshift/hello-openshift
+          env:
+            - name: RESPONSE
+              value: "Hello Serverless!"
+EOF
+
+cat <<-EOF | oc apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: curl
+  namespace: tenant-1
+  labels:
+    app: curl
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: curl
+  template:
+    metadata:
+      labels:
+        app: curl
+      annotations:
+        sidecar.istio.io/inject: 'true'
+    spec:
+      containers:
+      - name: curl
+        image: curlimages/curl
+        command:
+        - sleep
+        - "3600"
+EOF
+
+# Tenant 2
+cat <<-EOF | oc apply -f -
+apiVersion: serving.knative.dev/v1
```

Comment:
I wonder whether we should use `kn service create` instead as it's much smaller and possibly easier to understand ?

In this case it would be:

```
kn service create test-webapp -n tenant-2 \
    --annotation-service serving.knative.openshift.io/enablePassthrough=true \
    --annotation-revision sidecar.istio.io/inject=true \
    --env RESPONSE="Hello Serverless!" \
    --image docker.io/openshift/hello-openshift
```

Reply:
You could even combine that with a `for t in tenant-1 tenant-2; do` ... to reduce the boilerpate even further and create both services in one sweep.

Reply:
I would prefer to discuss the use kn as a separate topic. If we do this, we should do it over all the docs.

Reply:
In the upstream docs we often have tabs, explaining the different ways to do things (yaml, cli). However, I think this is possibly overkill here, but I really would love to see that this sections get a bit more condensed. Especially for thinks like smoke testing the CLI is much better suited than using temporary YAML files that you throw away later. I would definitely opt for pushing our user more to this kind of usage (that kind of interactive use case is what we made the CLI for). 

Ideally we should do this at other places, too, but I don't see it as a big issue if we have mixed usages, especially when one approach has a significant advantage of the other one.

Reply:
Sorry I'm still not convinced, IMHO:
- People need to have the `kn` client installed, I don't think it is safe to assume that cluster operators have that.
- We also create deployments, service, gateways, SMMR, SMCP and other resources with yaml so we still have that. I do not see much benefit in introducing another concept here.
- With YAML you directly see what you get (it does not hide anything) and can always put it directly in Git
- For the (hopefully) following Gitops deployment you need to understand the yamls (fields/annotations) anyways

I'm open to discuss this more broadly (with docs team), but this is a bit out of scope of that PR.

Reply:
I agree, that we should _not_ necessarily introduce `kn` here because of this doc change. But I still think that `kn` has its place especially for smoke tests like these as those smoke tests are rarely need to be tacked via GitOps. `kn` has some clear UI benefits here over using large YAML files that might even need to be modified for some fields (like the image reference).

So my suggestion here is to:

* Check whether the usage of tabs is possible and offer the `kn` solution as an alternative (I'm happy to contribute that part)
* but if not or only with a lot of effort, happy to stick to YAMLs only, but we might consider introducing `kn` more rigorously in a later step.